src/Entities/HangingEntity.cpp

```diff
@@ -29,3 +30,48 @@ void cHangingEntity::SpawnOn(cClientHandle & a_ClientHandle)
 
 
 
+
+void cHangingEntity::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
+{
+	Byte OppositeFacing = m_Facing + 2;
+
+	if (OppositeFacing >= 4)
+	{
+		OppositeFacing -= 4;
+	}
+
+	const auto & Pos = GetPosition();
+	const auto World = a_Chunk.GetWorld();
+	const auto SupportBlockPos = AddFaceDirection(Pos, ProtocolFaceToBlockFace(OppositeFacing));
+	BLOCKTYPE Block = World->GetBlock(SupportBlockPos);
```

Comment:
The block should be got from the chunk, and the world only fetched later (if required).

Reply:
What if the block the entity is hanging on isn't in the same chunk? Should I check that first instead?

Reply:
Ah, if you use the RelGetBlock function it will pointer-chase across chunks so there's no need to do bounds checking or worry about that. It should be much faster than looking up the block in the chunk map of the world.

Reply:
Oh! Great to know, thank you. I'll do that in the next patch